pkg/controller/daemon/controller.go

```diff
@@ -113,6 +114,9 @@ func NewDaemonSetsController(podInformer framework.SharedInformer, kubeClient cl
 	// TODO: remove the wrapper when every clients have moved to use the clientset.
 	eventBroadcaster.StartRecordingToSink(&unversionedcore.EventSinkImpl{Interface: kubeClient.Core().Events("")})
 
+	if kubeClient != nil && kubeClient.Core() != nil && kubeClient.Core().GetRESTClient() != nil {
```

Comment:
Done as much as possible. Doesn't work for Core (as we use clientset as an interface, and I'm not sure if it's possible to have a nil-implementations of interfaces).